pkg/meter/native/instruments.go

```diff
@@ -18,10 +18,76 @@
 // Package native provides a simple meter system for metrics. The metrics are aggregated by the meter provider.
 package native
 
-type nativeInstrument struct{}
+import (
+	"github.com/apache/skywalking-banyandb/banyand/queue"
+	"github.com/apache/skywalking-banyandb/pkg/meter"
+)
 
-func (nativeInstrument) Inc(_ float64, _ ...string)     {}
-func (nativeInstrument) Set(_ float64, _ ...string)     {}
-func (nativeInstrument) Add(_ float64, _ ...string)     {}
-func (nativeInstrument) Observe(_ float64, _ ...string) {}
-func (nativeInstrument) Delete(_ ...string) bool        { return false }
+type counter struct {
+	*metricVec
+}
+
+func newCounter(measureName string, pipeline queue.Client, scope meter.Scope) *counter {
+	return &counter{
+		newMetricVec(measureName, pipeline, scope),
+	}
+}
+
+func (c *counter) Inc(delta float64, labelValues ...string) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	key := generateKey(labelValues)
+	v, exist := c.metrics[key]
+	if !exist {
+		v = metricWithLabelValues{
+			labelValues: labelValues,
+		}
+	}
```

Comment:
Could you extract these lines to a method of metricVec, since all relevant methods will use it?